libnmstate/nm/profile.py

```diff
@@ -579,7 +587,10 @@ def classify_profile_for_actions(self, groups):
                             ACTION_DEACTIVATE_BEFOREHAND
                         )
                         groups["profiles_to_deactivate_beforehand"].add(self)
-                elif self.iface.type == InterfaceType.MAC_VLAN:
+                elif (
+                    self.iface.type == InterfaceType.MAC_VLAN
+                    or self.iface.type == InterfaceType.MAC_VTAP
+                ):
```

Comment:
A comment is required why and when it should be deactivated first.

And this seems not a good way to deactivate interface every time it has been included in desire state.

Reply:
I've added the comment and in addition I've added another condition to the statement. It is only being deactivated if the macvlan/macvtap interface is modified not only included in desired state.